### PR TITLE
#15 Scrapers for joker and Jpoint(2018-2019)

### LINKED ISSUE #15: Create Scrapy project
Basic scrapy project for extract data

### PR DESCRIPTION
**Pull request by AstralRomance**
Close #15 
Requirements in confeti_scraper/requirements.txt
Output jsons in confeti_scraper/output/valid_output
Run all scrapers and postprocessing with run_spiders.py

### CODE REVIEW

**Comment by marsofandrew** [comment]
As I know it's a compiled file `scraper/confeti_scraper/confeti_scraper/__pycache__/__init__.cpython-36.pyc`

**Comment by marsofandrew** [comment]
And also `scraper/confeti_scraper/confeti_scraper/__pycache__/settings.cpython-36.pyc` and the same `*3.pyc` file

**Review by marsofandrew — CHANGES_REQUESTED** [review]
See comments

**Comment by AstralRomance** [comment]
pycaches removed.

File: scraper/confeti_scraper/confeti_scraper/run_spiders.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+SCRAPERS = ['joker2019_scraper', 'jpoint2019_scraper', 'jpoint2018_scraper', 'joker2018_scraper']
+
+
+def run_scrapers():
+    for scraper in SCRAPERS:
+        os.system(f'scrapy crawl {scraper}  -o output/{scraper}.json')
+
+    postprocessing.run_postrpoc()
+
+run_scrapers()
```
Comment (marsofandrew) [inline scraper/confeti_scraper/confeti_scraper/run_spiders.py]: add new line

File: scraper/confeti_scraper/confeti_scraper/spiders/joker2018_scraper.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            speaker['name'] = speaker_sec.xpath('.//h5[@class="speaker-info_name"]/text()').get().strip()
+            speaker['avatar'] = speaker_sec.xpath('.//img[@class="img-fluid"]/@src').get()
+            speaker['bio'] = speaker_sec.xpath('.//div[@class="speaker-info_bio"]//p/text()').get().strip()
+            contact_info['company'] = (speaker_sec.xpath('.//h6[@class="speaker-info_company"]/text()').get(),str(*re.findall(r'\d\d\d\d',self.conference['name'])))
+            contact_info['twitterUsername'] =  speaker_sec.xpath('.//div[@class="speaker_profiles"]//a[@class="twitter_link"]/@href').get()
+            speaker['contactInfo']=contact_info
+            speakers_list.append(speaker)
+        report['speakers'] = speakers_list
+        self.conference['report'] = report
+        yield self.conference
```
Comment (marsofandrew) [inline scraper/confeti_scraper/confeti_scraper/spiders/joker2018_scraper.py]: add new line

File: scraper/confeti_scraper/scrapy.cfg [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# https://scrapyd.readthedocs.io/en/latest/deploy.html
+
+[settings]
+default = confeti_scraper.settings
+
+[deploy]
+#url = http://localhost:6800/
+project = confeti_scraper
+
+shell = ipython
```
Comment (marsofandrew) [inline scraper/confeti_scraper/scrapy.cfg]: add new line

File: scraper/confeti_scraper/confeti_scraper/spiders/jpoint_scraper.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            speaker['name'] = speaker_sec.xpath('.//h5[@class="speaker-info_name"]/text()').get().strip()
+            speaker['avatar'] = speaker_sec.xpath('.//img[@class="img-fluid"]/@src').get()
+            speaker['bio'] = speaker_sec.xpath('.//div[@class="speaker-info_bio"]//p/text()').get().strip()
+            contact_info['company'] = (speaker_sec.xpath('.//h6[@class="speaker-info_company"]/text()').get(),self.conference['year'])
+            contact_info['twitterUsername'] =  speaker_sec.xpath('.//div[@class="speaker_profiles"]//a[@class="twitter_link"]/@href').get()
+            speaker['contactInfo']=contact_info
+            speakers_list.append(speaker)
+        report['speakers'] = speakers_list
+        self.conference['report'] = report
+        yield self.conference
```
Comment (marsofandrew) [inline scraper/confeti_scraper/confeti_scraper/spiders/jpoint_scraper.py]: please add new line

**Review by marsofandrew — APPROVED** [review]

**Pull request merged**